Polyhedron/demo/Polyhedron/Plugins/Mesh_3/Mesh_3_plugin.cpp

```diff
@@ -692,6 +710,18 @@ void Mesh_3_plugin::mesh_3(const Mesh_type mesh_type,
       QMessageBox::critical(mw, tr(""), tr("ERROR: no data in selected item"));
       return;
     }
+#ifdef CGAL_USE_ITK
+    if ( sigma_weights > 0
+      && sigma_weights != image_item->sigma_weights())
+    {
+      image_item->set_image_weights(
+        CGAL::Mesh_3::generate_label_weights(*pImage, sigma_weights),
+        sigma_weights);
+    }
+#endif
```

Comment:
In the dialog, the section "Labelled Image" should be hidden, if the selection is not a label image (see how the section "Gray Level Image" is hidden, the same way.

It should also be hidden if `CGAL_USE_ITK` is not defined.